src/main/java/roboy/dialog/states/expoStates/DemonstrateAbilitiesState.java

```diff
@@ -0,0 +1,306 @@
+package roboy.dialog.states.expoStates;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import roboy.dialog.states.definitions.State;
+import roboy.dialog.states.definitions.StateParameters;
+import roboy.linguistics.Linguistics;
+import roboy.linguistics.sentenceanalysis.Interpretation;
+import roboy.ros.RosMainNode;
+import roboy.talk.PhraseCollection;
+import roboy.util.RandomList;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Implementations of Roboy's abilities.
+ *
+ * Following methods have to be implemented:
+ *  - wouldYouLikeToSeeDemoQuestions() -  provides a list of yes/no questions for the act method
+ *  - demonstrateAbility()             -  implementation of the ability (should block until demonstration is finished)
+ *  - afterDemoEndedPhrases()          -  provides a list of phrases that wrap up the ability demonstration
+ *
+ */
+enum RoboyAbility {
+
+//    SHAKE_HAND {
+//
```

Comment:
a lot of outcommented code here
if no one is going to implement it, better to remove the outcommented code

Reply:
planned for the future, i'd keep it

Reply:
ok